utils/python/CIME/XML/machines.py

```diff
@@ -175,8 +175,7 @@ def is_valid_compiler(self,compiler):
         """
         Check the compiler is valid for the current machine
 
-        >>> machobj = Machines()
```

Comment:
Thanks for cleaning this up. I forgot to update these tests when I changed the constructor.